src/vs/workbench/api/common/extHost.protocol.ts

```diff
@@ -385,7 +385,11 @@ export interface ILanguageConfigurationDto {
 	};
 }
 
-export type GlobPattern = string | { base: string; pattern: string; };
+export type GlobPattern = string | IRelativePattern;
+
+export interface IRelativePatternDto extends IRelativePattern {
+	baseUri: URI;
```

Comment:
You better use `UriComponents` because that's what will be received.

Reply:
Yup, pushed.